### PR TITLE
feat(circuit_conversions): allow NamedQubits in imported circuits

### PR DESCRIPTION
**Pull request by AthenaCaesura**
## Description

Users can now use NamedQubits in imported circuits. However, NamedQubits still cannot be used when importing single GatesOperations. I made this design choice because importing a GateOperation on a NamedQubit means mapping it to a number, which might be inconsistent between imports. 

For instance, a GateOperation on NamedQubit("a") could get the same label as NamedQubit("b") if they are translated a different time. Instead, users should translate the Gate and then pick a qubit themselves rather than translating a GateOperation with a NamedQubit.

## Please verify that you have completed the following steps

- [x] I have self-reviewed my code.
- [x] I have included test cases validating introduced feature/fix.
- [x] I have updated documentation.

### CODE REVIEW

File: src/orquestra/integrations/cirq/conversions/_circuit_conversions.py [hunk 1]
```diff
@@ -371,6 +371,7 @@ def _convert_gate_operation_to_orquestra(operation) -> _gates.GateOperation:
     if not all(isinstance(qubit, cirq.LineQubit) for qubit in operation.qubits):
         raise NotImplementedError(
             f"Failed to import {operation}. Only LineQubits are supported."
+            "for gates. Try converting the Gate and adding qubits after."
```
[the remark this answers is not in the document]
Comment (AthenaCaesura) [reply src/orquestra/integrations/cirq/conversions/_circuit_conversions.py:374]: For sure!

[the remark this answers is not in the document]
Comment (AthenaCaesura) [reply src/orquestra/integrations/cirq/conversions/_circuit_conversions.py]: Done!

File: tests/orquestra/integrations/cirq/conversions/circuit_conversions_test.py [hunk 2]
```diff
@@ -337,3 +337,27 @@ def test_with_cirq_only_gates_with_free_symbols_check(self, cirq_circuit):
     def test_with_unsupported_gates_raises_not_implemented_error(self, cirq_circuit):
         with pytest.raises(NotImplementedError):
             import_from_cirq(cirq_circuit)
+
+    def test_only_named_qubits_are_converted_to_labeled_qubit(self):
```
Comment (mstechly) [inline tests/orquestra/integrations/cirq/conversions/circuit_conversions_test.py:341]: Circuit in this test has only named qubits, so I think you switched the names of the tests?

Also, I'd like to see some edge cases like circuit with qubits:
[20, "b"], ["a", 0, "b"] (order shouldn't matter, right?)

**Review by mstechly — CHANGES_REQUESTED** [review]

File: tests/orquestra/integrations/cirq/conversions/circuit_conversions_test.py (hunk 2 above)
Comment (AthenaCaesura) [reply tests/orquestra/integrations/cirq/conversions/circuit_conversions_test.py:341]: nope! order shouldn't matter. Will add those test cases later tonight.

Comment (AthenaCaesura) [reply tests/orquestra/integrations/cirq/conversions/circuit_conversions_test.py:341]: Done!

**Review by mstechly — APPROVED** [review]
